File: app/src/main/java/by/egorgutko/autorization/data/AutorizationPreferenceInterfece.java
package by.egorgutko.autorization.data;

import android.content.Context;

import io.reactivex.Completable;
import io.reactivex.Single;

public interface AutorizationPreferenceInterfece {
    Completable addUserName(String name);
    Single<String> getUserName();
}

File: app/src/main/java/by/egorgutko/autorization/data/defaultPreference/AutorizationPreferenceInterfece.kt
package by.egorgutko.autorization.data.defaultPreference

import io.reactivex.Completable
import io.reactivex.Single

interface AutorizationPreferenceInterfece {
    //val userName: Single<String>
    fun getUserName(): Single<String>
    fun addUserName(name: String): Completable
}
